### PR TITLE
QuestionType tests and fixes

### PR DESCRIPTION
**Pull request by AdamFipke**
# Description

- Adds tests to questionType 
- Adds fixes to the questionType endpoint regarding these tests
- Added small fixes to CourseRolesGuard and RolesGuard
- Removed a redundant/unused endpoint "registerCourses" 

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update
- [ ] This requires a run of `yarn install`

# How Has This Been Tested?

image

Manually tested:
- Creating/deleting question types on both regular queues and the async question centre
- creating questions with the newly created question types

# Checklist:

- [x] I have performed a self-review of my own code
- [x] I have commented my code where needed 
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules
- [ ] Any UI changes have been checked to work on desktop, tablet, and mobile

### CODE REVIEW

File: packages/server/src/questionType/questionType.controller.ts [hunk 1]
```diff
@@ -13,54 +13,60 @@ import {
...
     } else {
-      res.status(400).send('Question already exists');
+      res.status(400).send('Question type already exists');
       return;
     }
   }
 
-  @Get(':c/:queueId')
+  @Get(':courseId/:queueId')
+  @UseGuards(QueueRolesGuard)
```
Comment (d3li0n) [inline packages/server/src/questionType/questionType.controller.ts:65]:
```suggestion
  @UseGuards(CourseRolesGuard, QueueRolesGuard)
```

File: packages/server/src/course/course.service.ts [hunk 2]
```diff
@@ -225,123 +225,6 @@ export class CourseService {
     }
   }
 
-  async registerCourses(
-    body: RegisterCourseParams[],
-    userId: number,
-  ): Promise<void> {
-    // obtains the ProfSectionGroupsModel of the professor
-    const profSectionGroups = await ProfSectionGroupsModel.findOne({
```
Comment (d3li0n) [inline packages/server/src/course/course.service.ts:233]: Can we remove any of unused imports?

File: packages/server/src/questionType/questionType.controller.ts [hunk 3]
```diff
@@ -13,54 +13,61 @@ import {
...
```
Comment (d3li0n) [inline packages/server/src/questionType/questionType.controller.ts:66]: Should it be a queue role guard or course role guard?

**Review by d3li0n — COMMENTED** [review]
Left some comments

File: packages/server/src/course/course.service.ts (hunk 2 above)
Comment (AdamFipke) [reply packages/server/src/course/course.service.ts:233]: Thanks, good catch lol

File: packages/server/src/questionType/questionType.controller.ts (hunk 3 above)
Comment (AdamFipke) [reply packages/server/src/questionType/questionType.controller.ts]: Actually you're right! good catch. When I was adding the guards, a part of me didn't realize that that one can have queueId be not a number. Good thing too, I had to make a new test for this and make some changes.

File: packages/server/src/course/course.controller.ts [hunk 4]
```diff
@@ -643,7 +642,7 @@ export class CourseController {
   }
 
   @Delete(':id/withdraw_course')
-  @UseGuards(JwtAuthGuard, EmailVerifiedGuard)
+  @UseGuards(JwtAuthGuard, CourseRolesGuard, EmailVerifiedGuard)
```
Comment (d3li0n) [inline packages/server/src/course/course.controller.ts:645]: Can you add roles decorator?

Comment (AdamFipke) [inline packages/server/src/course/course.controller.ts:645]:
```suggestion
  @UseGuards(JwtAuthGuard, CourseRolesGuard, EmailVerifiedGuard)
  @Roles(Role.Student, Role.PROFESSOR, Role.TA)
```

**Review by d3li0n — CHANGES_REQUESTED** [review]
One small change

**Review by d3li0n — APPROVED** [review]

**Pull request merged**